### ISSUE TITLE
[Feature Request] Finamp-specific volume control

### ISSUE THREAD

**Issue by Venryx** [jmshrv/finamp#500]
The songs in my library are apparently quite loud; to match the volume level I want, I need to put my system volume at ~5%.

This is a problem, because it means that then I cannot hear other apps with audio -- such as Google Maps for navigation, which I want to be able to hear alongside my music.

The solution would seem to be to add a volume control within Finamp's player specifically. Then I can reduce its volume enough that the system-level Media volume gives a good result for Finamp and Google Maps at the same time.

****

By the way, I notice there's a feature request for Replay Gain support: https://github.com/jmshrv/finamp/issues/468

That's good of course, but a volume control would still be needed to control the relative volume between Finamp and Google Maps. (or other use-cases, eg. android builds where it only provides ~10 system-level "volume ticks")

**Comment by Chaphasilor** [comment]
Would you want to adjust this volume frequently or could it be somewhere in the settings? 🤔

**Comment by Venryx** [comment]
For my particular use-case, having it in the settings would be "workable" -- although it does seem like something useful enough to be worth adding a little button for it. (eg. click the button, and it shows a vertical slider above it for volume)

**Comment by Chaphasilor** [comment]
I'm thinking about adding this as part of adding volume normalization (#468). Basically, when enabling volume normalization, you could set a "target volume" which all tracks should have (on average)

**Comment by Venryx** [comment]
> I'm thinking about adding this as part of adding volume normalization (#468). Basically, when enabling volume normalization, you could set a "target volume" which all tracks should have (on average)

That would be very nice indeed! Especially in contexts where ui interaction is difficult (eg. while driving), since it would (at least mostly) remove the need to change volumes for tracks.

**Comment by Venryx** [comment]
> Sweet. Also, another question: You mentioned that you can't hear Google Maps announcements while Finamp is playing. Are both apps running on the same device? Finamp _should_ "duck" the volume if navigation apps are playing audio, if that's not working we could probably fix this in an even better way.

Well the problem is that my music tracks are just too loud, so that I have to set the overall system volume at almost the lowest setting; so even though Finamp "ducks" the volume when Google Maps speaks, the system volume is so low that it still makes the Google Maps voice uncomfortably quiet.

I know I could fix this by pre-processing the tracks to be quieter, but it would be really nice to not have to do that. (which is why your suggestion above about a volume normalization feature sounds great to me)

**Comment by Chaphasilor** [comment]
Ah okay. But can you confirm that Finamp actually ducks, if your volume is high enough?

**Comment by Venryx** [comment]
> Ah okay. But can you confirm that Finamp actually ducks, if your volume is high enough?

Yep, confirmed; Finamp does quiet itself when the Google Maps voice speaks.

**Comment by Chaphasilor** [comment]
So audio normalization is working for me on Android already, and I've already added a setting that lets you adjust the target loudness. It's not perfect yet, but it should mostly work :)  
I could also easily add separate volume control, I'm just not sure *where* 😅

**Comment by Venryx** [comment]
> So audio normalization is working for me on Android already, and I've already added a setting that lets you adjust the target loudness. It's not perfect yet, but it should mostly work :) I could also easily add separate volume control, I'm just not sure _where_ 😅

Awesome!

As for placement of a volume control, could it not just be an icon-button next to the existing "heart" button at bottom-center?

Like this:
![](https://i.imgur.com/mbGMecL.jpeg)

Except with the heart-icon shifted to the left of course, to keep the heart and volume icons centered on-screen.

(If that's considered too ugly/out of place, then idk where; UI in general is fairly subjective, though that placement seems fine to me.)

**Comment by Chaphasilor** [comment]
Yeah, something like that might work. The button will have to be integrated into the redesign (#220), and there is a plan to prepare for controlling different media devices (e.g. speakers that you're casting to, once that is implemented). So ideally the volume panel would just have most of the interactions and UI elements needed for supporting more advanced volume control for different devices in the future.  
If I get around to designing some more mockups I might take a look at this problem :)

**Comment by felix920506** [comment]
Hi, I would also like to see this feature. Listening to my music library on my tablet (iPad Pro 2021) with my headphones, I would need to have the volume at the lowest step above 0 in order to not be deafened

**Comment by Reinachan** [comment]
As a desktop user, I'd like this feature as well. The thing is, I'm often listening to music while I'm in voice chat with a friend, as such I tend to need to adjust the volume fairly frequently. I would really want there to be an audio volume adjustment in the play area

**Comment by Chaphasilor** [comment]
FYI this feature dropped in 0.9.16 :)

**Comment by the-rene** [comment]
Is this configurable? I wondered, why the volume was at 100%, even if the system volume of that device was lower. And I think it is because of this change(?). If not, sorry for posting in an long closed issue.

I think it would be more intuitive, if the volume there is the same as the volume of the device playing: 

<img width="591" height="1240" alt="Image" src="https://github.com/user-attachments/assets/c9275923-bbd4-422a-ac94-f08b5e60aea1" />

I stumbled on this after trying the remote feature from #616 - because the similar function in Spotify controls the volume of the actual device playing. I think, that's the reason I find the current behavior unintuitive.

**Comment by felix920506** [comment]
@the-rene that volume being separate from the system volume is what the feature request is asking for. If you open something like VLC you will also find the volume controls like this.

**Comment by the-rene** [comment]
@felix920506 yes, I get that and get, that "volume being separate from the system volume" is a useful feature.

 On second thougt, just got confused, because it looks so native, that I expected to control the device volume. I looked a little around and think that this is what I mixed this up with the android-builin volume control: 

<img width="300"  alt="screenshot of android volume control" src="https://github.com/user-attachments/assets/1024497f-d880-4806-831e-e76f27011442" />


So, sorry for bringing this up here, I think this is perfectly valid and I only got confused.

**Comment by Chaphasilor** [comment]
@the-rene eventually I'd like to control the volume of remote devices as well, although in this case here you're just connected to Bluetooth. This will likely be added when we introduce support for controlling other Jellyfin clients via the "Play On" feature (i.e. controlling the Android TV app from Finamp). There it makes sense to control the volume of the actual playing device.  
But as Felix pointed out, the internal volume control for now is just to a) allow finer adjustments (i.e. if the system volume only has 20 volume steps, you can get finer control here), b) allow other Jellyfin clients to control Finamp's volume remotely, and c) allow independent volume adjustment on desktop platforms :)
